Add navigation and diagnostics docs to new format (#5)
wso2/docs-ballerina-visual-studio-code-plugin: en/docs/edit-the-code/code-navigation.md

> # Code Navigation

Reviewer: ```suggestion
# Code navigation
```

> ## Go to Definition 

Reviewer: ```suggestion
## Go to definition 
```

> For a symbol, this feature will navigate you to the definition of the particular symbol. For example, when invoking the `Go To Definition` option from a function call expression, you will be navigated to the definition of the function. Apart from jumping to the definition, the peek definition will also be supported. The behavior will be the same not only for the constructs within the sources in the current project but also for external modules and standard libraries as well.

Reviewer: ```suggestion
This feature navigates you to the definition of a particular symbol. For example, when you invoke the `Go To Definition` option on a function call expression, it navigates you to the definition of the function. Apart from that, the peek definition will also be supported. The behavior will be the same not only for the constructs within the sources in the current project but also for external modules and standard libraries as well.
```

<img src="https://github.com/wso2/ballerina-plugin-vscode/blob/main/resources/images/go-to-definition.gif?raw=true" width="100%" />

> ## Find all References

Reviewer: ```suggestion
## Find all references
```

Invoking the references on a symbol will prompt you with all the symbol references in the current project.

<img src="https://github.com/wso2/ballerina-plugin-vscode/blob/main/resources/images/find-all-references.gif?raw=true" width="100%" />

> ## Rename Symbols

Reviewer: ```suggestion
## Rename symbols
```

# Code navigation

> 

Reviewer 2: intro

Reviewer 2: A TW will add this.

## Go to definition 

> This feature navigates you to the definition of a particular symbol. For example, when you invoke the `Go To Definition` option on a function call expression, it navigates you to the definition of the function.

Reviewer 2: `when you invoke the `Go To Definition` option on a function call expression` is this done by clicking the function call expression and selecting the **Go To Definition** from the list? 
The giff was pretty fast.

Author: Same as above. VSCode users are familiar with how to go to the definition, `right click -> click 'Go to Definition'` or `CTRL/CMD+Click` shortcut.

Reviewer 2: A TW will add this by referring VS code docs

![Go to definition support](/en/docs/img/edit-the-code/code-navigation/go-to-definition.gif?raw=true)

## Go to symbol

> This feature allows you to navigate symbols inside a file.

Reviewer 2: Shall we explain in words how we actually do this? As I understood from [1] and from the giff, we need to use the `@` symbol to do this?

[1] https://code.visualstudio.com/docs/editor/editingevolved#_go-to-symbol

Author: This is a standard feature in VSCode. So the developers are usually familiar with the shortcut/approach of searching the symbols.

Reviewer 2: A TW will enhance these by referring the VS Code docs

![Go to symbol](/en/docs/img/edit-the-code/code-navigation/go-to-symbol.gif?raw=true)

## Find all references

> Invoking the references on a symbol will prompt you with all the symbol references in the current project.

Reviewer 2: Shall we explain this a bit more? how to do this really?

Reviewer 2: A TW will add this by referring the VS Code docs

![Find all references](/en/docs/img/edit-the-code/code-navigation/find-all-references.png?raw=true)

## Peek

> This feature allows you to peek the definition or references of a given symbol.

Reviewer 2: What do we mean by peek?

https://code.visualstudio.com/docs/editor/editingevolved#_peek

Reviewer 2: A TW will address this by referring the VS code docs.
